src/ensembl/src/shared/state/ens-object/ensObjectActions.ts

```diff
@@ -40,84 +40,99 @@ export const fetchEnsObjectAsyncActions = createAsyncAction(
-    } catch {
-      // FIXME: this is a temporary solution
-      response.track = builtTrackList(response);
+      const dispatchedPromise = dispatch(
+        getTrackPanelGene.initiate({
+          genomeId: payload.genomeId,
+          geneId: payload.objectId
+        })
+      );
+      const result = await dispatchedPromise;
+      dispatchedPromise.unsubscribe();
+
+      dispatch(fetchEnsObjectAsyncActions.request(ensObjectId));
+      const { genomeId } = payload;
+
+      const geneEnsObject = buildGeneObject({
+        objectId: buildEnsObjectId(payload),
+        genomeId,
+        gene: result.data?.gene
+      });
+
+      dispatch(
+        fetchEnsObjectAsyncActions.success({
+          id: ensObjectId,
+          data: geneEnsObject
+        })
+      );
+    } catch (error) {
+      dispatch(fetchEnsObjectAsyncActions.failure(error));
     }
+  };
 
-    dispatch(
-      fetchEnsObjectAsyncActions.success({
-        id: ensObjectId,
-        data: response
-      })
-    );
-  } catch (error) {
-    dispatch(fetchEnsObjectAsyncActions.failure(error));
-  }
-};
+export const fetchExampleEnsObjects =
+  (genomeId: string): ThunkAction<void, any, null, Action<string>> =>
+  async (dispatch, getState: () => RootState) => {
+    const state = getState();
+    const exampleFocusObjects = getGenomeExampleFocusObjects(state, genomeId);
 
-export const fetchExampleEnsObjects = (
-  genomeId: string
-): ThunkAction<void, any, null, Action<string>> => async (
-  dispatch,
-  getState: () => RootState
-) => {
-  const state = getState();
-  const exampleFocusObjects = getGenomeExampleFocusObjects(state, genomeId);
-
-  exampleFocusObjects.forEach(({ id, type }) => {
-    dispatch(fetchEnsObject({ genomeId, type, objectId: id }));
-  });
-};
+    exampleFocusObjects.forEach(({ id, type }) => {
+      dispatch(fetchEnsObject({ genomeId, type, objectId: id }));
+    });
+  };
+
+// FIXME: many fields here are unnecessary for an ensObject
+const buildGeneObject = (params: any): EnsObjectGene => {
+  const {
+    slice: {
+      location: { start, end },
+      region: { name: chromosome },
+      strand: { code: strand }
+    }
+  } = params.gene;
 
-// FIXME: this is a temporary solution, until the backend
-// fixes the api/object/track_list endpoint
-const builtTrackList = (ensObject: EnsObjectResponse) => {
   return {
-    additional_info: ensObject.bio_type || undefined,
-    description: ensObject.description,
-    ensembl_object_id: ensObject.object_id, // we don't use this field
-    label: ensObject.label,
-    track_id: TrackId.GENE,
-    child_tracks: [],
-    stable_id: ensObject.stable_id
+    type: 'gene',
+    object_id: params.objectId,
+    genome_id: params.genomeId,
+    label: params.gene.symbol,
+    location: {
+      chromosome,
+      start,
+      end
+    },
+    stable_id: params.gene.unversioned_stable_id,
+    versioned_stable_id: params.gene.stable_id,
+    bio_type: '',
```

Comment:
Eventually, ensObject should not store biotype data; the component that needs it should get it directly from the fetched gene. But since I have left this change for a subsequent PR, I have added the biotype back to the ens object for the time being.